devdoc/rc_ptr.md

```diff
@@ -0,0 +1,41 @@
+# `rc_ptr` requirements
+
+## Overview
+
+`rc_ptr` is a module that encapsulates reference-counted pointers.
+
+## Exposed API
+```c
+typedef void (*RC_PTR_FREE_FUNC)(void* context);
+
+typedef struct RC_PTR_TAG
+{
+    const void* ptr;
+    RC_PTR_FREE_FUNC free_func;
+} RC_PTR;
+
+THANDLE_TYPE_DECLARE(RC_PTR);
+
+#define PRI_RC_PTR "p"
+
+#define RC_PTR_VALUE(rc) ((rc)->ptr)
+
+#define RC_PTR_AS_VOIDPTR(rc) (((rc) == NULL) ? NULL : (rc)->ptr)
+
+MOCKABLE_FUNCTION(, THANDLE(RC_PTR), rc_ptr_create_with_move_pointer, void*, ptr, RC_PTR_FREE_FUNC, free_func);
```

Comment:
Because we don't know the type of the data. 

Doing this would require templatizing the entire module on the type of data: `THANDLE(CHANNEL(DATA))` 

Can be done, but I haven't written a lot of macro code and don't want to spend time on it right now. Priorities.